src/components/themed/SimpleTextInput.tsx

```diff
@@ -324,11 +348,55 @@ const InputField = styledWithRef(AnimatedTextInput)<{
 })
 
 const TouchContainer = styled(TouchableOpacity)(theme => ({
-  // Increase tappable area with padding, while net 0 with negative margin to visually appear as if 0 margins/padding
   padding: theme.rem(1),
   margin: -theme.rem(1)
 }))
 
+const CancelButton = styled(Animated.View)<{
+  focusAnimation: SharedValue<number>
+}>(theme => ({ focusAnimation }) => {
+  const width = theme.rem(4)
+  return [
+    {
+      marginLeft: theme.rem(0.5),
+      justifyContent: 'center',
+      alignItems: 'center',
+      height: theme.rem(3.5)
+    },
+    useAnimatedStyle(() => {
+      'worklet'
+      return {
+        width: interpolate(focusAnimation.value, [0, 1], [0, width]),
```

Comment:
We shouldn't animate layout props because of performance issue. We want to use transform instead to make our animations smooth. If you can't figure out how we can animate this component in a performant way, then we should just not animate the cancel button.